include/mbgl/style/sources/image_source.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <mbgl/style/source.hpp>
+#include <mbgl/util/image.hpp>
+#include <mbgl/util/optional.hpp>
+
+namespace mbgl {
+class LatLng;
+class AsyncRequest;
+
+namespace style {
+
+class ImageSource : public Source {
+public:
+    ImageSource(std::string id, const std::vector<LatLng>);
```

Comment:
@incanus - thats an excellent point. I've tried to improve that in the SDKs in the [draft API PR](https://github.com/mapbox/mapbox-gl-native/pull/9110). If the Quad struct makes sense there, I might even move it into native to avoid confusion